src/components/TwoFactorAuthModal.js

```diff
@@ -64,15 +61,17 @@ class TwoFactorAuthModal extends React.Component {
       closeTwoFactorAuthModalWindow,
       modal: {mode},
       disableTFA,
-      confirm2FA
+      confirm2FA,
+      updateProfile,
     } = this.props;
     const { currentCode } = this.state;
     try {
       mode === 'disable' ?
-        await disableTFA(currentCode) :
-        await confirm2FA(currentCode);
+        await disableTFA(currentCode).then(data => updateProfile(data.user)) :
```

Comment:
1. Business logic in component should be hidden inside of `disableTFA`.
2. Functions `disableTFA` and `confirm2FA` are using different postfixes. Decide to use `TFA` or `2FA`.